docker/lib/dependabot/docker/update_checker.rb

```diff
@@ -316,7 +316,7 @@ def prerelease?(tag)
       def numeric_version_from(tag)
         return unless tag.match?(NAME_WITH_VERSION)
 
-        tag.match(NAME_WITH_VERSION).named_captures.fetch("version").downcase
+        tag.match(NAME_WITH_VERSION).named_captures.fetch("version").downcase.gsub(/_/i, ".")
```

Comment:
Probably a better place to handle this change would be in the `version.rb` file, as has been done for other languages like Gradle, but I wouldn't know how to do that.

Reply:
Thank you @RodrigoPetter! I didn't see your PR in time and created a separate one at #5734. Coincidentally, I used the approach you mentioned here used for other ecosystems 👍.

Anyways, this PR was very useful and I reused part of the specs and added attribution to you in the commit ❤️.

Reply:
Wow @deivid-rodriguez, thank you! I'm glad I could help. 

Probably your PR fixes the original problem that I was facing. My PR doesn't seems necessary anymore.

I would prefer that, for more complex cases of non semver, it would be taken into consideration the cases in the issues #2692 and #4329.

With that in mind, I'm closing my PR.

Reply:
Thank you! We'll try to keep improving things and you're welcome to keep sending us patches. We'll try to be more responsive next time!